core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/waiters/WaiterExecutorHelper.java

```diff
@@ -32,12 +34,20 @@
  */
 @SdkInternalApi
 public final class WaiterExecutorHelper<T> {
+    private static final int DEFAULT_MAX_ATTEMPTS = 3;
+    private static final BackoffStrategy DEFAULT_BACKOFF_STRATEGY = FixedDelayBackoffStrategy.create(Duration.ofSeconds(2));
     private final List<WaiterAcceptor<? super T>> waiterAcceptors;
-    private final PollingStrategy pollingStrategy;
+    private final BackoffStrategy backoffStrategy;
+    private final Duration maxWaitTime;
+    private final int maxAttempts;
 
-    public WaiterExecutorHelper(List<WaiterAcceptor<? super T>> waiterAcceptors, PollingStrategy pollingStrategy) {
+    public WaiterExecutorHelper(List<WaiterAcceptor<? super T>> waiterAcceptors,
+                                WaiterOverrideConfiguration overrideConfiguration) {
         this.waiterAcceptors = waiterAcceptors;
-        this.pollingStrategy = pollingStrategy;
+        this.backoffStrategy = overrideConfiguration.backoffStrategy().orElse(DEFAULT_BACKOFF_STRATEGY);
+        this.maxWaitTime = overrideConfiguration.maxWaitTime().orElse(null);
+        this.maxAttempts = overrideConfiguration.maxAttempts().orElse(DEFAULT_MAX_ATTEMPTS);
```

Comment:
This is a little confusing. If the customer specifies the maxWaitTime, they might not know to jack up the maxAttempts, which are likely less time consuming to execute than their maxWaitTime if they stuck with the default values.

I suppose this is no different than the requestTimeout and maxRetries, though. Perhaps we can rename maxWaitTime to waitTimeout to draw the parallel between the RetryPolicy.maxAttempts/WaiterOverrideConfig.maxAttempts and ClientOverrideConfig.requestTimeout/WaiterOverrideConfig.waitTimeout?

Reply:
I'd expect most of the customers to use the generated waiters w/o any configurations. Yeah, it might be a bit confusing for customers who do want to customize it.

+1 rename maxWaitTimeout to waitTimeout.